src/js/media-theme-element.js

```diff
@@ -60,6 +60,20 @@ export class MediaThemeElement extends window.HTMLElement {
     })
 
     this.createRenderer();
+
+    // In case the template prop was set before custom element upgrade.
+    // https://web.dev/custom-elements-best-practices/#make-properties-lazy
+    this.#upgradeProperty('template');
```

Comment:
question(non-blocking): In the best practices example referenced by the link ☝️, this happens in `connectedCallback`. Any reason why we're doing this in the `constructor` instead?

Reply:
good call, I think I figured because it worked and there was no need to re-evaluate it on each connectedCallback but thinking about this again connectedCallback is called slightly later and it's needed there to safely access custom element attributes which we do in the props getter so I'll move it here 👍